src/main/java/info/movietrash/cinemabase/service/impl/MovieServiceImpl.java

```diff
@@ -0,0 +1,53 @@
+package info.movietrash.cinemabase.service.impl;
+
+import info.movietrash.cinemabase.converter.MovieConverter;
+import info.movietrash.cinemabase.dto.MovieDto;
+import info.movietrash.cinemabase.exception.ErrorMessages;
+import info.movietrash.cinemabase.model.Movie;
+import info.movietrash.cinemabase.model.User;
+import info.movietrash.cinemabase.repository.MovieRepository;
+import info.movietrash.cinemabase.service.MovieService;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+public class MovieServiceImpl implements MovieService {
+
+    private final MovieRepository movieRepository;
+    private final MovieConverter movieConverter;
+
+    @Override
+    public Long addMovie(MovieDto movieDto) {
+        return movieRepository.save(movieConverter.toModel(movieDto)).getId();
```

Comment:
add check on existing movie before trying to persist it

Reply:
Query(select m from Movie m where m.externalId = :externalId") should be added to MovieRepository

Reply:
done